query/src/lib.rs

```diff
@@ -44,6 +44,20 @@ pub type SrcVarName = String;          // Do not include the required syntactic
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
 pub struct Variable(pub PlainSymbol);
 
+impl Variable {
+    pub fn as_str(&self) -> &str {
```

Comment:
`Into` throughout?  Generally, I find named functions easier to follow, but Rust style is definitely to use `Into`.